tests/controller/datacontroller.js

```diff
@@ -293,26 +337,57 @@ describe( 'DataController', () => {
 	} );
 
 	describe( 'stringify', () => {
-		it( 'should get paragraph with text', () => {
+		beforeEach( () => {
 			modelDocument.schema.registerItem( 'paragraph', '$block' );
 			modelDocument.schema.registerItem( 'div', '$block' );
-			const modelElement = parse( '<div><paragraph>foo</paragraph></div>', modelDocument.schema );
 
 			buildModelConverter().for( data.modelToView ).fromElement( 'paragraph' ).toElement( 'p' );
+		} );
+
+		it( 'should stringify a content of an element', () => {
+			const modelElement = parseModel( '<div><paragraph>foo</paragraph></div>', modelDocument.schema );
 
 			expect( data.stringify( modelElement ) ).to.equal( '<p>foo</p>' );
 		} );
+
+		it( 'should stringify a content of a document fragment', () => {
+			const modelDocumentFragment = parseModel( '<paragraph>foo</paragraph><paragraph>bar</paragraph>', modelDocument.schema );
+
+			expect( data.stringify( modelDocumentFragment ) ).to.equal( '<p>foo</p><p>bar</p>' );
+		} );
 	} );
 
 	describe( 'toView', () => {
-		it( 'should get view element P with text', () => {
+		beforeEach( () => {
 			modelDocument.schema.registerItem( 'paragraph', '$block' );
 			modelDocument.schema.registerItem( 'div', '$block' );
-			const modelElement = parse( '<div><paragraph>foo</paragraph></div>', modelDocument.schema );
 
 			buildModelConverter().for( data.modelToView ).fromElement( 'paragraph' ).toElement( 'p' );
+		} );
+
+		it( 'should convert a content of an element', () => {
+			const modelElement = parseModel( '<div><paragraph>foo</paragraph></div>', modelDocument.schema );
+
+			const viewDocumentFragment = data.toView( modelElement );
+
+			expect( viewDocumentFragment ).to.be.instanceOf( ViewDocumentFragment );
+
+			const viewElement = viewDocumentFragment.getChild( 0 );
+
+			expect( viewElement.name ).to.equal( 'p' );
+			expect( viewElement.childCount ).to.equal( 1 );
+			expect( viewElement.getChild( 0 ).data ).to.equal( 'foo' );
+		} );
+
+		it( 'should convert a document fragment', () => {
+			const modelDocumentFragment = parseModel( '<paragraph>foo</paragraph><paragraph>bar</paragraph>', modelDocument.schema );
+
+			const viewDocumentFragment = data.toView( modelDocumentFragment );
+
+			expect( viewDocumentFragment ).to.be.instanceOf( ViewDocumentFragment );
+			expect( viewDocumentFragment ).to.have.property( 'childCount', 2 );
 
-			const viewElement = data.toView( modelElement ).getChild( 0 );
+			const viewElement = viewDocumentFragment.getChild( 0 );
```

Comment:
I mean – there are a lot more things which could be broken below, but it'd be unwise to try to test them so late. I'm checking the number of children and the first child to verify that enough elements were converted and that they make some sense.